src/core/loop_sccdag_attributes/include/noelle/core/SCCDAGAttrs.hpp

```diff
@@ -97,6 +98,8 @@ class SCCDAGAttrs {
    */
   LoopCarriedVariable *checkIfReducible(SCC *scc, LoopForestNode *loop);
 
+  std::vector<Value *> checkIfPeriodic(SCC *scc, LoopForestNode *loopNode);

```

Comment:
changed in [6a6b8e4](https://github.com/arcana-lab/noelle/pull/74/commits/6a6b8e4baaf30d0006471d6dbc48a2409462a7c4). if i use a tuple, i wasn't sure how to return an empty tuple when the SCC is not periodic, because the empty tuple wouldn't match the function return type `std::tuple<Value*, Value*, Value>`. my current solution is to just pass in `nullptr`s